File: 1306-minimum-absolute-difference/1306-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        int n = arr.length;
        
        // Sort the array first
        Arrays.sort(arr);

        // Find the minimum absolute difference
        int minAbsDiff = Integer.MAX_VALUE;
        for (int i = 1; i < n; i++) {
            minAbsDiff = Math.min(minAbsDiff, arr[i] - arr[i-1]);
        }

        // Create the result list
        List<List<Integer>> ans = new ArrayList<>();

        // Add pairs with the minimum absolute difference
        for (int i = 1; i < n; i++) { // Start from i = 1 to avoid out-of-bounds
            if (arr[i] - arr[i-1] == minAbsDiff) {
                ans.add(Arrays.asList(arr[i-1], arr[i]));
            }
        }

        return ans;
    }
}

File: 1306-minimum-absolute-difference/1306-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        /*
        [4,2,1,3]
        */
        ios_base::sync_with_stdio(false);  
        cin.tie(NULL);

        int n = arr.size();
        int minDiff = INT_MAX;
        vector<vector<int>> ans;
        sort(arr.begin(),arr.end());

        for(int i = 0; i < n-1; i++)
        {
            minDiff = min(minDiff, arr[i+1]-arr[i]);
        }

        for(int i = 0; i < n-1; i++){
            if(minDiff == (arr[i+1]-arr[i]))
            {
                ans.push_back({arr[i],arr[i+1]});
            }
        }
        return ans;
    }
};